MonoGame.Framework/Graphics/Vertices/VertexBuffer.cs

```diff
@@ -26,10 +26,6 @@ protected VertexBuffer(GraphicsDevice graphicsDevice, VertexDeclaration vertexDe
             this.VertexCount = vertexCount;
             this.BufferUsage = bufferUsage;
 
-            // Make sure the graphics device is assigned in the vertex declaration.
-            if (vertexDeclaration.GraphicsDevice != graphicsDevice)
-                vertexDeclaration.GraphicsDevice = graphicsDevice;

```

Comment:
XNA even uses these as a `static`...

https://msdn.microsoft.com/en-us/library/microsoft.xna.framework.graphics.vertexpositioncolor.vertexdeclaration.aspx

So not sure hope they ensure it is cleaned up and reused when a GraphicsDevice is disposed and recreated.